### PR TITLE
Feat/react 18

### PR DESCRIPTION
**Pull request by FIZLIN**

### CODE REVIEW

File: packages/app/src/scenes/App.tsx [hunk 1]
```diff
@@ -99,6 +104,26 @@ const App: FC = () => {
...
+  // // FIXME: Default url
+  // if (pathname === ROUTES.base) {
+  //   return (
+  //     <Routes>
+  //       <Route path="*" element={<Navigate to={ROUTES.explore} replace />} />
+  //     </Routes>
+  //   );
+  // }
+
+  // // PUBLIC ROUTES
```
Comment (PAKovalev) [inline packages/app/src/scenes/App.tsx]: Why is it commented?

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: packages/app/src/core/utils/router.utils.tsx [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
-import {SwitchProps} from 'react-router';
+import {matchPath, Navigate, Route, Routes} from 'react-router-dom';
+import {RoutesProps} from 'react-router';
 
 import {RouteConfigInterface} from 'core/interfaces';
 
 export const isTargetRoute = (currentPath: string, routes: RouteConfigInterface[]): boolean => {
   return routes.some((route) => {
-    return !!matchPath(currentPath, {path: route.path, exact: route.exact});
+    return !!matchPath({path: route.path}, currentPath);
```
Comment (dmitry-yudakov) [inline packages/app/src/core/utils/router.utils.tsx]: Is it clear what replaces now missing `exact`? Or are we supposed to use something like `/a/b/*`? It doesn't feel right to just drop it.

File: packages/app/src/scenes/App.tsx [hunk 3]
```diff
@@ -49,23 +52,25 @@ const App: FC = () => {
...
-          text={
-            configLoadingErrorCode === httpErrorCodes.MAINTENANCE
-              ? t('systemMessages.underMaintenance')
-              : t('errors.somethingWentWrongTryAgain')
-          }
-          showRefreshButton
-          theme={themeStore.theme}
-        />
-      </ThemeProvider>
+      <>
```
Comment (dmitry-yudakov) [inline packages/app/src/scenes/App.tsx]: Is this extra `<></>` needed?

File: packages/app/src/scenes/App.tsx (hunk 1 above)
Comment (FIZLIN) [reply packages/app/src/scenes/App.tsx]: Can up when rebasing the onboarding changes. I've removed the comment.

**Review by PAKovalev — APPROVED** [review]

File: packages/core/package.json [hunk 4]
```diff
@@ -15,11 +15,12 @@
...
-    "react-dom": "^16.14.0",
-    "mobx-state-tree": "^5.1.3",
-    "detect-browser": "^5.2.0",
+    "axios": "^0.22.0",
     "date-fns-tz": "^1.3.4",
-    "axios": "^0.22.0"
+    "detect-browser": "^5.2.0",
+    "mobx-state-tree": "^5.1.3",
+    "react": "^18.2.0",
+    "react-com": "^1.0.6",
```
Comment (dmitry-yudakov) [inline packages/core/package.json]: `react-com` seems like a mistake?

File: packages/core/package.json [hunk 5]
```diff
@@ -15,11 +15,12 @@
...
+    "react-dom": "^16.14.0"
```
Comment (dmitry-yudakov) [inline packages/core/package.json]: `react-dom` should be `^18.2.0`

Comment (FIZLIN) [reply packages/core/package.json]: ✅

File: packages/app/src/scenes/App.tsx (hunk 3 above)
Comment (FIZLIN) [reply packages/app/src/scenes/App.tsx]: ✅

**Pull request merged**